citest/gcp_testing/gcp_agent.py

```diff
@@ -0,0 +1,287 @@
+    """Determine variable bindings to pass to a method.
+
+    Args:
+      method: [string] The name of the desired method (in the resource_type).
+      resource_type: [string] The resource type operating on.
+      resource_id: [string] The particular instance to operate on, if any.
+      kwargs: Additional variable bindings
+
+    Returns
+      kwargs ammended with additional default variables, if applicable.
+    """
+    discovery_doc = self.discovery_document
+    info = discovery_doc.get('resources', {}).get(resource_type, None)
+    if info is None:
+      raise KeyError('Unknown {0} Resource type={1}'
+                     .format(discovery_doc['name'].title(), resource_type))
+    method_spec = info.get('methods', {}).get(method, None)
+    if method_spec is None:
+      raise KeyError('Unknown method={0} on resource={1}'
+                     .format(method, resource_type))
+    parameters = method_spec.get('parameters', {})
+    result = {key: value for key, value in kwargs.items() if value is not None}
+    resource_id_param = None
+    if resource_id is not None:
+      resource_id_param = method_spec['parameterOrder'][-1]
+      result[resource_id_param] = resource_id
+
+    missing = []
+    for key, key_spec in parameters.items():
+      if key not in kwargs and key != resource_id_param:
+        def_value = self.__default_variables.get(key, None)
+        if def_value is not None:
+          result[key] = def_value
+        elif key_spec.get('required', False):
+          missing.append(key)
+
+    if missing:
+      raise ValueError('"{0}()" is missing required parameters: {1}'
+                       .format(method, missing))
+    return result
+
+  def get_resource(self, resource_type, resource_id=None, **kwargs):
+    """Get instance metadata details.
+
+    Args:
+      resource_type: [string] The type of the resource instance to get.
+      resource_id: [string] The id of the resource instance to get.
+      kwargs: [kwargs] Additional parameters may be required depending
+         on the resource type (such as zone, etc).
+    """
+    return self.invoke_resource(
+        'get', resource_type=resource_type, resource_id=resource_id, **kwargs)
+
+  def invoke_resource(self, method, resource_type, resource_id=None, **kwargs):
+    """Invoke a method on a resource type or instance.
+
+    Args:
+      method: [string] The operation to perform as named under the
+          |resource_type| in the discovery document.
+      resource_type: [string] The type of the resource instance to operate on.
+      resource_id: [string] The id of the resource instance, or None to operate
+        on the resource type or collection.
+      kwargs: [kwargs] Additional parameters may be required depending
+         on the resource type and method.
+    """
+    variables = self.resource_method_to_variables(
+        method, resource_type, resource_id=resource_id, **kwargs)
+
+    resource_obj = vars(self.__service).get(resource_type, None)
```

Comment:
You can replace the `get` on the `vars` dict with regular dict access: `vars(thing)[resource_type]`.